### PR TITLE
initial implementation for pixels with streaming

### PR DESCRIPTION
**Pull request by erinaldidb**
To enable the streaming capability a parameter is required in catalog reading step.

```python
from dbx.pixels import Catalog

catalog = Catalog(spark, table)
catalog_df = catalog.catalog(path, streaming=True)
```

### CODE REVIEW

**Comment by dmoore247** [comment]
Note that test and lint failed.
@erinaldidb Please run `make style` and update PR.

File: dbx/pixels/catalog.py [hunk 1]
```diff
@@ -61,26 +68,93 @@ def __init__(self, spark, table: str = "main.pixels_solacc.object_catalog"):
...
+            .drop("content")
+            )
 
-    def catalog(self, path: str, pattern: str = "*", recurse: bool = True) -> DataFrame:
+    def catalog(self, 
+                path: str, 
+                pattern: str = "*", 
+                recurse: bool = True, 
+                streaming: bool = False, 
+                streamCheckpointPath: str = "/tmp/pixels_checkpoints/",
```
Comment (dmoore247) [inline dbx/pixels/catalog.py:96]: The checkpoint path should be a volume and a parameter.

**Review by dmoore247 — CHANGES_REQUESTED** [review]
Excellent work!!
Make checkpoint a parameter, a volume.
Please add a python 'unit' test if at all possible (runs with `make test`)

File: tests/dbx/test_stream.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+def setup(spark: SparkSession):
+    spark.sql("CREATE DATABASE IF NOT EXISTS main.pixels_solacc")
+    yield
+
+    try:
+        checkpoints = list(dbutils.fs.ls(CHECKPOINT_BASE_PATH))
+        if len(checkpoints) > 0:
+            dbutils.fs.rm(CHECKPOINT_BASE_PATH, True)
+    except DatabricksError as err:
+        if "No file or directory exists on path" in str(err):
```
Comment (dmoore247) [inline tests/dbx/test_stream.py:34]: Will error return the path that was attempted or do we have to provide the context?

**Review by dmoore247 — APPROVED** [review]
LGTM!
See in-line question.

File: tests/dbx/test_stream.py (hunk 2 above)
Comment (erinaldidb) [reply tests/dbx/test_stream.py:34]: Yes, it will also return the path you requested to list from.

**Pull request merged**